### PR TITLE
feat: add C implementation for `math/base/special/boxcox1pinv`

### PR DESCRIPTION
**Pull request by Userf1995**
## Description

> What is the purpose of this pull request?

This pull request:

-   add C implementation for math/base/special/boxcox1pinv

## Related Issues

> Does this pull request have any related issues?

This pull request:

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

**Comment by Rejoan-Sardar** [comment]
Something happend on these PR that's some file not add,I am working on it.

File: lib/node_modules/@stdlib/math/base/special/boxcox1pinv/README.md [hunk 1]
```diff
@@ -115,6 +115,99 @@ for ( i = 0; i < y.length; i++ ) {
...
+
+    double out;
+    int i;
+    int j;
+    for ( i = 0; i < 3; i++ ) {
+        for ( j = 0; j < 3; j++ ){
+            out = stdlib_base_boxcox1pinv( x[ i ], y[ j ] );
+            printf ( "y: %lf, x: %lf, out: %lf\n", x[ i ], y[ j ], out );
+        }
+    }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/boxcox1pinv/README.md]:
```suggestion
    const double y[] = { -1.0, 10.0, 1.0 };
    const double l[] = { -0.5, 5.0, 0.5 };

    double b;
    int i;
    int j;
    for ( i = 0; i < 3; i++ ) {
        for ( j = 0; j < 3; j++ ){
            b = stdlib_base_boxcox1pinv( y[ i ], l[ j ] );
            printf ( "boxcox1pinv(%lf, %lf) = %lf\n", y[ i ], l[ j ], b );
        }
    }
```

File: lib/node_modules/@stdlib/math/base/special/boxcox1pinv/examples/c/example.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	double out;
+	int i;
+	int j;
+	for ( i = 0; i < 3; i++ ) {
+		for ( j = 0; j < 3; j++ ){
+			out = stdlib_base_boxcox1pinv( x[ i ], y[ j ] );
+			printf ( "y: %lf, x: %lf, out: %lf\n", x[ i ], y[ j ], out );
+		}
+	}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/boxcox1pinv/examples/c/example.c]:
```suggestion
	const double y[] = { -1.0, 10.0, 1.0 };
	const double l[] = { -0.5, 5.0, 0.5 };

	double b;
	int i;
	int j;
	for ( i = 0; i < 3; i++ ) {
		for ( j = 0; j < 3; j++ ){
			b = stdlib_base_boxcox1pinv( y[ i ], l[ j ] );
			printf ( "boxcox1pinv(%lf, %lf) = %lf", y[ i ], l[ j ], b );
		}
	}
```

File: lib/node_modules/@stdlib/math/base/special/boxcox1pinv/manifest.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      "include": [
+        "./include"
+      ],
+      "libraries": [],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/special/abs",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/math/base/special/log1p"
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/boxcox1pinv/manifest.json]:
```suggestion
        "@stdlib/math/base/special/log1p",
        "@stdlib/math/base/special/expm1"
```

File: lib/node_modules/@stdlib/math/base/special/boxcox1pinv/test/test.native.js [hunk 4]
```diff
@@ -0,0 +1,277 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/boxcox1pinv/test/test.native.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/boxcox1pinv/examples/c/example.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	const double y[] = { -1.0, 10.0, 1.0 };
+	const double l[] = { -0.5, 5.0, 0.5 };
+
+	double b;
+	int i;
+	int j;
+	for ( i = 0; i < 3; i++ ) {
+		for ( j = 0; j < 3; j++ ){
+			b = stdlib_base_boxcox1pinv( y[ i ], l[ j ] );
+			printf ( "boxcox1pinv(%lf, %lf) = %lf", y[ i ], l[ j ], b );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/boxcox1pinv/examples/c/example.c]:
```suggestion
			printf ( "boxcox1pinv(%lf, %lf) = %lf\n", y[ i ], l[ j ], b );
```

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @Rejoan-Sardar !

**Pull request merged**